### ISSUE TITLE
[clang-format] QualifierOrder ignores `// clang-format off`

### ISSUE THREAD

**Issue by remyjette** [llvm/llvm-project#53643]
Trying out the release/14.x branch this morning

```
$ ./clang-format --version
clang-format version 14.0.0 (https://github.com/llvm/llvm-project.git c0fde7580aac12143d63291af452e261808dc4ec)

$ cat test.cpp
// clang-format off
static inline int a = 0;
// clang-format on
int main() {}

$ clang-format -style="{QualifierAlignment: Custom, QualifierOrder: ['inline', 'static', 'type']}" test.cpp
// clang-format off
inline static int a = 0;
// clang-format on
int main() {}
```

Expected result: The file is unchanged.

Actual result: Qualifiers are ordered by `QualifierOrder` despite being in a `// clang-format off` block.

**Comment by mkurdej** [comment]
Revision URI: https://reviews.llvm.org/D119218

**Comment by mkurdej** [comment]
> Would anyone mind if we cherry pick this into the 14.0 branch, I'm seeing Qualifier ordering creeping into people .clang-format being used on GitHub projects, and I think we need this escape hatch?

+1

Go for it, it can't do any harm and it will avoid people creating issues for a thing that's already fixed.

**Comment by mydeveloperday** [comment]
@tstellar I've probably broken your beautiful system... but I'm trying to backport to 14. I think it was cherry picking against the main branch and this caused an empty commit.. is there any written documentation on the correct way to do this. (I don't think the cherry pick should have failed I think its very isolated, @mkurdej don't you think?)

**Comment by mydeveloperday** [comment]
@tstellar ok, maybe its running now, so do I have to add the "release label" as well as the "che**y-pick" comment so it knows which branch to backport it to?...

Absolutely awesome BTW...

**Comment by tstellar** [comment]
@mydeveloperday It pulls the branch name from the milestone, so you need to set that first.

**Comment by mydeveloperday** [comment]
@tstellar once the check on the PR are complete what do we need to do to get it added to the branch?

https://github.com/llvmbot/llvm-project/pull/48

Should I simply approve the PR?

**Comment by tstellar** [comment]
@mydeveloperday I will review and merge it.

**Comment by tstellar** [comment]
Merged: 6e7a837fdd55 67e0b64c0e6c
